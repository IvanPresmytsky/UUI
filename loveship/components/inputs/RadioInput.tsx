import { withMods } from '@epam/uui-core';
import { RadioInput as uuiRadioInput, RadioInputProps as UuiRadioInputProps, RadioInputMods as UuiRadioInputMods } from '@epam/uui';
import css from './RadioInput.module.scss';

export interface RadioInputMods extends UuiRadioInputMods {
    /**
     * @default 'light'
     */
    theme?: 'light' | 'dark';
}

export interface RadioInputProps extends RadioInputMods, UuiRadioInputProps {}

function applyRadioInputMods(mods: RadioInputProps) {
    return [mods.theme === 'dark' && css.themeDark];
}

export const RadioInput = withMods<UuiRadioInputProps, RadioInputMods>(uuiRadioInput, applyRadioInputMods);
