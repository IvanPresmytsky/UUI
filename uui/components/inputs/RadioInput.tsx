import { withMods } from '@epam/uui-core';
import { RadioInput as uuiRadioInput, RadioInputProps as uuiRadioInputProps } from '@epam/uui-components';
import { ReactComponent as RadioPoint } from '@epam/assets/icons/radio_dot-fill.svg';
import { settings } from '../../settings';
import css from './RadioInput.module.scss';

type RadioInputMods = {
    /**
     * Defines component size.
     * @default '18'
     */
    size?: '12' | '18';
};

/** Represents the properties of a RadioInput component. */
export type RadioInputProps = RadioInputMods & uuiRadioInputProps;

function applyRadioInputMods(mods: RadioInputProps) {
    return [
        css.root,
        css['size-' + (mods.size || settings.sizes.defaults.radioInput)],
        'uui-radio-input-container',
        'uui-color-primary',
    ];
}

export const RadioInput = withMods<uuiRadioInputProps, RadioInputProps>(
    uuiRadioInput,
    applyRadioInputMods,
    (props) => ({ icon: props.icon ? props.icon : RadioPoint }),
);
