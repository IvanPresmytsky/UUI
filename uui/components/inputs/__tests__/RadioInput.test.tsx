import React from 'react';
import { RadioInput } from '../RadioInput';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('RadioInput', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<RadioInput value={ null } onValueChange={ jest.fn } />);
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<RadioInput value={ null } onValueChange={ jest.fn } size="18" label="Open" />);
        expect(tree).toMatchSnapshot();
    });
});
