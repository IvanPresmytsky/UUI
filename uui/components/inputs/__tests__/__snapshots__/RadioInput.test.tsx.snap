// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioInput should be rendered correctly with maximum 1`] = `
<label
  className="container uui-disabled uui-readonly uui-invalid root size-18 uui-radio-input-container uui-color-primary"
>
  <div
    className="uui-radioinput"
  >
    <input
      aria-checked={false}
      aria-disabled={true}
      aria-readonly={true}
      aria-required={true}
      checked={false}
      disabled={true}
      readOnly={true}
      required={true}
      type="radio"
    />
  </div>
  <div
    className="uui-input-label"
  >
    Open
  </div>
</label>
`;

exports[`RadioInput should be rendered correctly with minimum props 1`] = `
<label
  className="container root size-18 uui-radio-input-container uui-color-primary -clickable"
>
  <div
    className="uui-radioinput"
  >
    <input
      aria-checked={false}
      checked={false}
      onChange={[Function]}
      type="radio"
    />
  </div>
</label>
`;
